### PR TITLE
[Docs] Update drop table behavior in spark-ddl docs

### LINKED ISSUE #5646: Documented drop table behavior for Spark is outdated.
### Apache Iceberg version

0.14.0 (latest release)

### Query engine

Spark

### Please describe the bug 🐞

#3056 added support for `PURGE` flag to Spark.
However, the corresponding `DROP TABLE` doc is outdated: https://iceberg.apache.org/docs/latest/spark-ddl/#drop-table

### PR DESCRIPTION
**Pull request by sumeetgajjar**
#3056 added support for `PURGE` flag to Spark.
However, the corresponding `DROP TABLE` doc is outdated.
This PR aims at documenting the updated behavior of the `DROP TABLE` DDL command.

Closes #5646

### CODE REVIEW

**Comment by sumeetgajjar** [comment]
A gentle ping @Fokko @samredai

File: docs/spark-ddl.md [hunk 1]
```diff
@@ -132,12 +132,28 @@ The new table properties in the `REPLACE TABLE` command will be merged with any
 
 ## `DROP TABLE`
 
-To delete a table, run:
+The drop table behavior changed in 0.14.
+
+Prior to 0.14, running `DROP TABLE` would remove the table from the catalog and delete the table contents as well.
+
+From 0.14 onwards, `DROP TABLE` would only remove the table from the catalog.
+In order to delete the table contents `DROP TABLE PURGE` should be used.
```
Comment (wypoon) [inline docs/spark-ddl.md:140]: Is it worth mentioning that gc.enabled has to be true for the contents to be deleted?

**Review by wypoon — APPROVED** [review]

File: docs/spark-ddl.md (hunk 1 above)
Comment (sumeetgajjar) [reply docs/spark-ddl.md:140]: I believe it is implied that if an operation involves removing any table files then `gc.enabled` should be true.
With that said, I didn't find an occurrence of `gc.enabled` anywhere in the iceberg spark docs. Thus decided to follow the same pattern.

File: docs/spark-ddl.md [hunk 2]
```diff
@@ -132,12 +132,28 @@ The new table properties in the `REPLACE TABLE` command will be merged with any
...
+
+To drop the table from the catalog, run:
 
 ```sql
 DROP TABLE prod.db.sample
 ```
 
+### `DROP TABLE PURGE`
+
+To delete a table, run:
```
Comment (samredai) [inline docs/spark-ddl.md:152]: To make the distinction a bit clearer with `DROP TABLE`, how about rewording this to:

> To drop the table from the catalog and delete the table's contents, run:

**Review by samredai — APPROVED** [review]
LGTM, I just left a small suggestion. 👍

File: docs/spark-ddl.md (hunk 2 above)
Comment (sumeetgajjar) [reply docs/spark-ddl.md:152]: Done.

Comment (sumeetgajjar) [reply docs/spark-ddl.md:152]: Hi @samredai, thank you for your comment, took care of it in the latest commit.

**Comment by sumeetgajjar** [comment]
Hi @wypoon @samredai, thank you for your comments and review on this PR.
can you also review the exact same change for the 0.14 branch: https://github.com/apache/iceberg/pull/5647?

**Comment by sumeetgajjar** [comment]
@pvary @flyrain @Fokko 
Since this PR is already reviewed, can you please help us merge this it?

**Comment by sumeetgajjar** [comment]
Thanks @pvary for the prompt response!!!
Should we merge the cherry-pick of this fix to the 0.14 branch as well https://github.com/apache/iceberg/pull/5647?

**Comment by sumeetgajjar** [comment]
Thank you @pvary for merging this PR and @wypoon and @samredai for the review!
